frappe/permissions.py

```diff
@@ -525,7 +526,7 @@ def allow_everything():
 def get_allowed_docs_for_doctype(user_permissions, doctype):
 	'''Returns all the docs from the passed user_permission
 		that are allowed under provide doctype'''
-	return [d.get('doc') for d in user_permissions if not d.get('applicable_for') or d.get('applicable_for') == doctype]
+	return [d.get('doc') for d in user_permissions if (not d.get('applicable_for') or d.get('applicable_for') == doctype) and (d.get('is_default') or len(user_permissions) == 1)]
```

Comment:
Check for default should not be here... this function should return the name of all allowed docs(because there are cases where we need all the allowed docs from this method. like in `permissions.py`)

Based on the number of returned doc you can choose the default doc. eg. If only one docnname is returned then that doc can be set as default otherwise you need to choose the default one based on the is_default flag.